src/domains/queries.ml

```diff
@@ -329,3 +311,29 @@ struct
 
   let hash x = 31 * order x + hash_arg x
 end
+
+
+let eval_int_binop (module Bool: Lattice.S with type t = bool) binop (ask: ask) e1 e2: Bool.t =
+  let t1 = Cilfacade.typeOf e1 in
+  let t2 = Cilfacade.typeOf e2 in
+  let (_, e) = Cilfacade.doBinOp binop e1 t1 e2 t2 in
```

Comment:
What does `doBinOp` do in case the expressions have different type?

Reply:
The whole point of `doBinOp` is that it does the exact same thing as the C parser: it finds the common type and inserts the required casts. The logic for that is rather involved (as currently can be seen from my copy-paste to `Cilfacade`). This was also required to fix some incompatible ikinds errors that otherwise arise from just putting two incompatible types of integers into a `BinOp`.